Python/symtable.c

```diff
@@ -2440,18 +2486,52 @@ symtable_visit_params(struct symtable *st, asdl_arg_seq *args)
 }
 
 static int
-symtable_visit_annotation(struct symtable *st, expr_ty annotation)
+symtable_visit_annotation(struct symtable *st, expr_ty annotation,
+                          struct _symtable_entry *parent_ste, void *key)
 {
     int future_annotations = st->st_future->ff_features & CO_FUTURE_ANNOTATIONS;
-    if (future_annotations &&
-        !symtable_enter_block(st, &_Py_ID(_annotation), AnnotationBlock,
-                              (void *)annotation, annotation->lineno,
-                              annotation->col_offset, annotation->end_lineno,
-                              annotation->end_col_offset)) {
-        VISIT_QUIT(st, 0);
+    if (future_annotations) {
+        if(!symtable_enter_block(st, &_Py_ID(_annotation), AnnotationBlock,
+                                 key, LOCATION(annotation))) {
+            VISIT_QUIT(st, 0);
+        }
+    }
+    else {
+        if (st->st_cur->ste_annotation_block == NULL) {
```

Comment:
I had to check myself that `st->st_cur` here and `parent_ste` on line 2506 were the same object. Would it be clearer to use `parent_ste` here as well?